packages/core/src/route/types.ts

```diff
@@ -84,12 +90,19 @@ export interface IPropDecoratorOptions {
 	typeFactory?: TypeValueFactory;
 	required?: boolean;
 }
-
-export type IPropDecoratorOptionsFactory = () => Maybe<IPropDecoratorOptions>;
-
-export interface IPropDecoratorMetadata {
-	key: string;
-	optsFactory?: IPropDecoratorOptionsFactory;
+export type IPropDecoratorOptionsFactory = IDecoratorOptionsFactory<IPropDecoratorOptions>;
+export type IPropDecoratorMetadata = IDecoratorMetadata<IPropDecoratorOptions>;
+
+export type IPropDependsDecoratorOptions = string[];
+export type IPropDependsDecoratorOptionsFactory = IDecoratorOptionsFactory<IPropDependsDecoratorOptions>;
+export type IPropDependsDecoratorMetadata = IDecoratorMetadata<IPropDependsDecoratorOptions>;
+
+export interface IDefinitionDecoratorOptions {
```

Comment:
probably we could use some existing types: https://github.com/kogosoftwarellc/open-api/blob/master/packages/openapi-types/index.ts#L640